### ISSUE TITLE
[FR] play in floating window

### ISSUE THREAD

**Issue by ildar** [sschueller/peertube-android#154]
as other video player do: NewPipe, VLC,...

**Comment by dhk2** [comment]
Got this working on my test system. There's some Android version specific code that will require more testing on various versions though. Adding a button was being difficult so I put switching to floating mode on a longpress of the full screen button.

**Comment by sschueller** [comment]
> Got this working on my test system. There's some Android version specific code that will require more testing on various versions though. Adding a button was being difficult so I put switching to floating mode on a longpress of the full screen button.

If this functions like the YouTube app with background play I would consider leaving the feature on permanently.

**Comment by dhk2** [comment]
makes sense.

**Comment by dhk2** [comment]
I'm about 80% of the way there. In order to keep orphaned floating windows from happening when opening different videos I had to change the #launchmode of videoplayactivity in the manifest from "singletop" to "singletask".  Everything seems to be working properly after the change. I can't see a need for multiple instances of the video player activity, but I might be missing something.

**Comment by dhk2** [comment]
Of course, after creating the pull request I figured out that switching it to singleinstance allows me to fix the problem where the back button went back to home screen instead of video list after restoring from PIP mode.
If the video list is destroyed while in PIP mode it will reopen the default view on backpress now. Youtube is able to keep the full back stack due to using fragments so the fragment manager keeps track even if the activity goes away. That would require a degree of re-architecting I'm not ready for yet :)
I think I've figured out how the get the video to play at the bottom of the screen without using PIP mode, now to see if I can duplicate it.
